src/main/java/org/springframework/integration/dsl/config/IntegrationFlowBeanPostProcessor.java

```diff
@@ -143,6 +160,7 @@ private Object processStandardIntegrationFlow(StandardIntegrationFlow flow, Stri
 					endpointBeanName = generateBeanName(endpoint);
 				}
 				registerComponent(endpoint, endpointBeanName, beanName, registerSingleton);
+				integrationComponents.set(i, endpoint);
```

Comment:
I use this fix to avoid extra `BeanFactory` access. Since `StandardIntegrationFlow` can be used as a container for its components. That was some idea to go ahead with the scope and even avoid bean registration for `IntegrationFlow` components. 
We still can do that, though, if you are interested. For example in case of `IntegrationFLowContext` we can initialize and start everything but we can bypass beans registration.
Although it can be useful only over `IntegrationFLowContext`...